### PR TITLE
changed the gradient input and corresponding grad

### PR DESCRIPTION
**Pull request by Rafael-G-C**

### CODE REVIEW

File: jax_test3.py [hunk 1]
```diff
@@ -1,57 +1,117 @@
...
+    # these will keep track whether we are on the x y or z coordinate
+    i_xyz = 0
+    a_xyz = 0
+    for i in range(2, len(state)):
+        state[i] += 1  # change the item state
+        for a in range(2, len(state)):
+            state[a] += 1
+            eval_E = derv(
+                E_NN,
+                [charges.size, charges, *(coordinates.reshape(3 * natoms).tolist())],
```
Comment (robertodr) [inline jax_test3.py]: I think the `tolist` is not actually necessary and just using `*coordinates.reshape(3*natoms)` will work too. Note that the reshaping can be done once outside the loop and it should help with efficiency:
```python
flat_coords = coordinates.reshape(3*natoms)
for i in range(2, len(state)):
     ...
```

**Review by robertodr — APPROVED** [review]
Very good! Great that you fixed the analytical Hessian.

File: jax_test3.py [hunk 2]
```diff
@@ -1,57 +1,117 @@
...
+    return e_nn
 
-grad_E = get_hessian(jnp.array([[1.0,2.0,3.0],[2.0,1.0,3.0],[3.0,1.0,2.0]]),[1.0,1.0,1.0])
+
+def get_grad(coordinates, charges):
+    gradient = []
+    natoms = len(charges)
+    state = np.zeros(
+        2 + 3 * natoms, dtype=int
+    )  # makes an array that derv can work with
```
Comment (robertodr) [inline jax_test3.py:63]:
```suggestion
    )  # makes an array that derv can work with
    flat_coords = coordinates.reshape(3 * natoms)
```

File: jax_test3.py [hunk 3]
```diff
@@ -1,57 +1,117 @@
...
+    for item in range(2, len(state)):
+        state[item] = 1  # change the array to the wanted derivation for the hessian
+        eval_E = derv(
+            E_NN,
+            [charges.size, charges, *(coordinates.reshape(3 * natoms).tolist())],
```
Comment (robertodr) [inline jax_test3.py]:
```suggestion
            [charges.size, charges, *flat_coords],
```

File: jax_test3.py [hunk 4]
```diff
@@ -1,57 +1,117 @@
...
+# this will generate the full hessian
+def get_hessian(coordinates, charges):
+    natoms = len(charges)
+    state = np.zeros(2 + 3 * natoms, dtype=int)
+    hessian = np.zeros(
+        (natoms, 3) * 2
+    )  # reshaping the hessian gave errors so here's created a priori
+    # these will keep track whether we are on the x y or z coordinate
+    i_xyz = 0
+    a_xyz = 0
```
Comment (robertodr) [inline jax_test3.py:89]:
```suggestion
    a_xyz = 0
    flat_coords = coordinates.reshape(3 * natoms)
```

File: jax_test3.py (hunk 1 above)
Comment (robertodr) [inline jax_test3.py]:
```suggestion
                [charges.size, charges, *flat_coords],
```

**Review by robertodr — COMMENTED** [review]
Few changes: the reshaping of `coordinates` can be done once and outside the loop(s)
